core/reducers/filter.js

```diff
@@ -0,0 +1,70 @@
+import {
+  COMPONENTS_FILTER_ADD_VALUE, COMPONENTS_FILTER_REMOVE_VALUE, COMPONENTS_FILTER_CLEAR_VALUES,
+  BLUEPRINTS_FILTER_ADD_VALUE, BLUEPRINTS_FILTER_REMOVE_VALUE, BLUEPRINTS_FILTER_CLEAR_VALUES
+} from '../actions/filter';
+
+const filtersWithNewFilter = (filters, filterValue) => {
+  if (filters.length === 0) {
+    return [...filters, filterValue]
+  } else {
+    const index = filters.findIndex(filter => filter.key === filterValue.key);
+    return [
+      ...filters.slice(0, index),
+      ...filters.slice(index + 1),
+      filterValue
+    ]
+  }
+};
+
+const filtersWithRemovedFilter = (filters, filterValue) => {
+  const index = filters.indexOf(filterValue);
+  return [
+    ...filters.slice(0, index),
+    ...filters.slice(index + 1)
+  ];
+};
+
+const updateFilters = (state, filters, filterList) => {
+  return Object.assign({}, state,
+    { [filterList]: Object.assign({}, state[filterList],
+      { filterValues: filters })
+    }
+  );
+};
+
+const handlers = {
+
+  [COMPONENTS_FILTER_ADD_VALUE]: (state, action) => {
+    return updateFilters(state, filtersWithNewFilter(state.components.filterValues, action.payload.filter), 'components')
+  },
+
+  [COMPONENTS_FILTER_REMOVE_VALUE]: (state, action) => {
+    return updateFilters(state, filtersWithRemovedFilter(state.components.filterValues, action.payload.filter), 'components')
+  },
+
+  [COMPONENTS_FILTER_CLEAR_VALUES]: (state) => {
+    return updateFilters(state, [], 'components')
+  },
+
+  [BLUEPRINTS_FILTER_ADD_VALUE]: (state, action) => {
+    return updateFilters(state, filtersWithNewFilter(state.components.filterValues, action.payload.filter), 'blueprints')
+  },
+
+  [BLUEPRINTS_FILTER_REMOVE_VALUE]: (state, action) => {
+    return updateFilters(state, filtersWithRemovedFilter(state.components.filterValues, action.payload.filter), 'blueprints')
+  },
+
+  [BLUEPRINTS_FILTER_CLEAR_VALUES]: (state) => {
+    return updateFilters(state, [], 'blueprints')
+  },
+};
```

Comment:
@jkozol The feedback I had received from one of the UXD js devs was to use this pattern instead of a `switch` statement. But when he shared it out with the rest of the team, one of the other devs shared [this example from another product](https://github.com/ManageIQ/miq_v2v_ui_plugin/blob/master/app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardClustersStep/MappingWizardClustersStepReducer.js) which is using it. 

I'm happy to change this back to `switch` which seems to be the standard that everyone's using. But I wanted to give you a chance to review and let me know what you think.

Reply:
Thanks for sharing this. Did he explain why he prefers this pattern over switch statements? I don't have a particular preference and just used switches because that's what is mainly used in the redux documentation.

Reply:
Some of the comments regarding `switch`:

> switch/case is just a weird exception to the normal JavaScript pattern of "code blocks are in curly brackets"
because it's sometimes useful to match multiple patterns and run both branches of code
so braces become less useful there.
But instead people use it as a replacement for `if() { } else if() { } else { }`
because zero braces is better than 6 right?
it's also more confusing. So-- functions. no if-else-if-else, but clear lines between what runs and what doesn't
I am surprised it is not more popular-- people like the syntax that involves the least typing.
I believe `switch` should probably only be used when functions don't work as well.. if you only want exactly one case to run, then it should just be a function in a lookup table